### PR TITLE
corrections and additions for calculator

### PR DESCRIPTION
**Pull request by TimZib**

### CODE REVIEW

File: stage2-calculator/calculator.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  /**
+   *
+   *
+   * @param {Number} a
+   * @param {Number} b
+   * @return {Number} the sum of a + b
+   * @memberof Calculator
+   */
+  addMethod(a, b) {
+    if (typeof a === 'number' && typeof a === 'number') {
```
Comment (rimus) [inline stage2-calculator/calculator.js:22]: `typeof a === 'number'` is duplicated here. Parameter `b` is not checking.

File: stage2-calculator/calculator.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  /**
+   *
+   *
+   * @param {Number} a
+   * @param {Number} b
+   * @return {Number} the product of a * b
+   * @memberof Calculator
+   */
+  multiplyMethod(a, b) {
+    if (typeof a === 'number' && typeof a === 'number') {
```
Comment (rimus) [inline stage2-calculator/calculator.js:38]: `typeof a === 'number'` is duplicated here. Parameter `b` is not checking.

File: stage2-calculator/add-multiply.spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const addition = [
+    { a: 8, b: 8, sum: 16 },
+    { a: 'twelve', b: 6, sum: 18 },
+    { a: 5, b: '74', sum: 79 },
+    { a: '&', b: 26, sum: 29 },
+    { a: 11, b: 'zero', sum: 11 },
+    { a: 9, b: 7, sum: 16 },
+    { a: true, b: 8, sum: 32 },
+    { a: 54, b: 3, sum: 57 },
+    { a: 7, b: Number, sum: 46 },
```
Comment (rimus) [inline stage2-calculator/add-multiply.spec.js:15]: Are all of your tests passing? :)

File: stage2-calculator/add-multiply.spec.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  const multiplication = [
+    { a: 'eight', b: 8, sum: 64 },
+    { a: 12, b: 6, sum: 72 },
+    { a: 5, b: false, sum: 370 },
+    { a: 3, b: 26, sum: 78 },
+    { a: 11, b: 0, sum: 'null' },
+    { a: 9, b: 7, sum: 63 },
+    { a: 24, b: 8, sum: -192 },
+    { a: 54, b: String, sum: 162 },
+    { a: 7, b: 39, sum: 273 },
```
Comment (rimus) [inline stage2-calculator/add-multiply.spec.js:35]: Are all of your tests passing? :)

**Review by rimus — CHANGES_REQUESTED** [review]
Also I haven't seen negative cases.

File: stage2-calculator/calculator.js (hunk 1 above)
Comment (TimZib) [inline stage2-calculator/calculator.js:22]: That was a typo. 
if (typeof a === 'number' && typeof b === 'number')

File: stage2-calculator/calculator.js (hunk 2 above)
Comment (TimZib) [reply stage2-calculator/calculator.js:38]: That was a typo. 
if (typeof a === 'number' && typeof b === 'number')

File: stage2-calculator/calculator.js (hunk 1 above)
Comment (TimZib) [reply stage2-calculator/calculator.js:22]: That was a typo. 
if (typeof a === 'number' && typeof b === 'number')

File: stage2-calculator/add-multiply.spec.js (hunk 4 above)
Comment (TimZib) [reply stage2-calculator/add-multiply.spec.js:35]: Not all. All in one. Negative and positive scenarios.

File: stage2-calculator/add-multiply.spec.js (hunk 3 above)
Comment (TimZib) [reply stage2-calculator/add-multiply.spec.js:15]: Not all. All in one. Negative and positive scenarios.

**Review by TimZib — COMMENTED** [review]
Some answers

File: stage2-calculator/calculator.js (hunk 1 above)
Comment (rimus) [reply stage2-calculator/calculator.js:22]: Why it still not fixed here? :)

File: stage2-calculator/add-multiply.spec.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  const sumMultError = () => calc.multiplyMethod('/', 2);
+  it('should return Error', function () {
+    expect(sumMultError).to.throw(Error);
+  });
+
+  const ssumMultError = () => calc.multiplyMethod(false, 4);
+  it('should return Error', function () {
+    expect(ssumMultError).to.throw(Error);
+  });
```
Comment (rimus) [inline stage2-calculator/add-multiply.spec.js:56]: These it-blocks also can be put into sngle one with `forEach()`

**Review by rimus — CHANGES_REQUESTED** [review]

**Review by rimus — APPROVED** [review]
Looks good.

**Pull request merged**